code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -398,6 +398,13 @@
 			M.reagents.remove_reagent(R.type,1)
 	..()
 
+/datum/reagent/medicine/charcoal/reaction_mob(mob/living/carbon/M, method=TOUCH, reac_volume)
+	if(method == INGEST)
+		return
+	M.reagents.remove_reagent(/datum/reagent/medicine/charcoal/, reac_volume) //We really should not be injecting an insoluble granular material.
```

Comment:
Question.

If someone takes a 50u charcoal pill, then gets 50u of charcoal splashed on them, won't this empty their bloodstream of charcoal?

Reply:
Yes it probably will. I think i will move all the reagent conversion stuff to its own on_transfer proc to avoid this kind of stuff.